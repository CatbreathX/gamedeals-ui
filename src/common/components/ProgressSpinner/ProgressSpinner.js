import {
  CircularProgressSpinner,
  ProgressSpinnerBox,
  ProgressSpinnerContainer,
} from 'common/components/ProgressSpinner/layout/styles';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';

export const ProgressSpinner = ({ delay }) => {
  const timerRef = React.useRef();
  const [canDisplayProgress, setCanDisplayProgress] = useState(false);

  useEffect(() => {
    timerRef.current = setTimeout(() => {
      setCanDisplayProgress(true);
    }, delay);
    return () => clearTimeout(timerRef.current);
  }, []); '';

  return (
    <>
      {canDisplayProgress &&
        <ProgressSpinnerContainer>
          <ProgressSpinnerBox>
            <CircularProgressSpinner />
          </ProgressSpinnerBox>
        </ProgressSpinnerContainer>
      }

      {!canDisplayProgress && null}
    </>
  );
};

ProgressSpinner.propTypes = {
  delay: PropTypes.number,
};

ProgressSpinner.defaultProps = {
  delay: 1,
};
