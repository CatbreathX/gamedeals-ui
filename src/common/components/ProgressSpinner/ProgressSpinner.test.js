import { render, screen } from '@testing-library/react';
import { ProgressSpinner } from 'common/components/ProgressSpinner/ProgressSpinner';

describe('ProgressSpinner', () => {
  test('should render correctly', () => {
    const {asFragment} = renderComponent();
    expect(asFragment()).toMatchSnapshot();
  })
  test('should show progress spinner', () => {
    renderComponent();
    expect(screen.getByRole('progressbar')).toBeInTheDocument();
  });
});

const renderComponent = (props = {}) => {
  return render(<ProgressSpinner {...props} />);
};
