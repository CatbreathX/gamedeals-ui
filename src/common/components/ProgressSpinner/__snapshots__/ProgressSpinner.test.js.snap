// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressSpinner should render correctly 1`] = `
<DocumentFragment>
  <div
    class="euwrsax2 MuiBox-root css-1j8js41-ProgressSpinnerContainer"
  >
    <div
      class="euwrsax1 MuiBox-root css-5sxe48-ProgressSpinnerBox"
    >
      <span
        class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary euwrsax0 css-a1sm72-MuiCircularProgress-root-CircularProgressSpinner"
        role="progressbar"
        style="width: 40px; height: 40px;"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;
